### PR TITLE
ACMS-1494: acquia_cms_headless: PHPUnit testcases for Hybrid Mode.

### PR DESCRIPTION
**Pull request by deepakmishra2**
**Motivation**
Fixes ACMS-1494

**Proposed changes**
PHP unit test for headless in hybrid mode

### CODE REVIEW

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    // Test if Enable headless mode checkbox works.
+    $this->drupalGet('/admin/tour/dashboard');
+    $checkbox = $assert_session->elementExists('css', '#edit-headless-mode');
+    $this->assertNotEmpty($checkbox);
+    $checkbox->check();
+    $page->pressButton('Save');
+
+    // Test Headless Dashboard link is not available.
+    $toolbar = $assert_session->elementExists('css', '#toolbar-administration');
+    $tour_link = $assert_session->elementExists('named', ['link', 'Acquia CMS Wizard'], $toolbar);
```
Comment (apathak18) [inline modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:46]: @deepakmishra2 any reason to not to make $tour_link variable as a camel case ?

cc: @vishalkhode1

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessBrowserTestBase.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace Drupal\Tests\acquia_cms_headless\Functional;
+
+use Acquia\DrupalEnvironmentDetector\AcquiaDrupalEnvironmentDetector;
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Base class for the HeadlessDashboard Browser tests.
+ */
+class HeadlessBrowserTestBase extends BrowserTestBase {
```
Comment (apathak18) [inline modules/acquia_cms_headless/tests/src/Functional/HeadlessBrowserTestBase.php:11]: **HeadlessBrowserTestBase** looks identical to **DashboardBrowserTestBase** class only difference is **DashboardSectionTrait** in the **DashboardBrowserTestBase** class

@deepakmishra2 Can we remove DashboardBrowserTestBase as only class DashboardApiUsersTest extends DashboardBrowserTestBase?
Instead we can extended HeadlessBrowserTestBase and include the trait DashboardSectionTrait within DashboardApiUsersTest?

cc: @vishalkhode1 @rajeshreeputra @chandan-singh7929

Comment (deepakmishra2) [reply modules/acquia_cms_headless/tests/src/Functional/HeadlessBrowserTestBase.php:11]: @apathak18 No we cannot do that because going forward we will be having more headless test [ACMS-1484, ACMS-1481, ACMS-1485, etc] in which we will be using HeadlessBrowserTestBase and not DashboardBrowserTestBase.

Comment (apathak18) [reply modules/acquia_cms_headless/tests/src/Functional/HeadlessBrowserTestBase.php:11]: With the Merge of PR https://github.com/acquia/acquia_cms/pull/1419/files#diff-c3c58804ab9ff9eef2745abf8d76c4975b02062d7527ed69e0eadf58bf778874R11
@deepakmishra2 I guess you need to rebase and remove this file.

cc: @chandan-singh7929 @vishalkhode1

**Comment by chandan-singh7929** [comment]
> @chandan-singh7929 Make variable names in same format throughout within the method i.e **testAcmsWizardMenu**

Done

**Review by rajeshreeputra — COMMENTED** [review]
please rebase branch

**Review by rajeshreeputra — APPROVED** [review]
LGTM

File: modules/acquia_cms_headless/tests/src/Functional/DashboardApiUsersTest.php [hunk 3]
```diff
@@ -60,7 +60,7 @@ public function testSection(): void {
 
     // Test table body exist and has data in same order.
     $this->assertEquals('Headless', $this->getTableBodyColumn(0)->getText());
-    $this->assertEquals('Headless Role', $this->getTableBodyColumn(1)->getText());
+    $this->assertEquals('Headless Administrator', $this->getTableBodyColumn(1)->getText());
```
Comment (apathak18) [inline modules/acquia_cms_headless/tests/src/Functional/DashboardApiUsersTest.php:63]: @chandan-singh7929 Rebase from feature/headless_test_coverage and revert this change
This change is already pushed into PR: https://github.com/acquia/acquia_cms/pull/1431/files#diff-5570e81b9d7033df76b458d3155cf5546824e4b1f5be5b9dd984006cdd5b2da7R63

Comment (chandan-singh7929) [reply modules/acquia_cms_headless/tests/src/Functional/DashboardApiUsersTest.php]: Done

**Review by apathak18 — APPROVED** [review]
LGTM

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ * @group acquia_cms_headless
+ * @group low_risk
+ */
+class HeadlessHybridModeTest extends HeadlessTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'acquia_cms_toolbar',
```
Comment (vishalkhode1) [inline modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:17]: @deepakmishra2 This should be either integrated tests or we should skip this tests if module: acquia_cms_toolbar is not present or maybe we should people like other drupal.org modules i.e adding optional module dependencies in require-dev section ?
cc: @chandan-singh7929  / @rajeshreeputra

Comment (deepakmishra2) [reply modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:17]: @vishalkhode1 - we had a long discussion among team members to either make it integrated or require acquia_cms_toolbar to check for menu. So we finalized to use the module instead making it integrated test. Let me know your thoughts @chandan-singh7929  @rajeshreeputra

Comment (vishalkhode1) [reply modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php]: If not integrated, then let's either module dependencies in dev or enable module through code and test if module exist and is enabled, so that in isolation it doesn't break. See example [here](https://github.com/acquia/acquia_cms/blob/develop/modules/acquia_cms_search/tests/src/FunctionalJavascript/AcquiaFacetSearchTest.php#L118).

Comment (deepakmishra2) [reply modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php]: done

**Review by vishalkhode1 — APPROVED** [review]
